libraries/ms-common/test/test_x86_uart.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import socket
+import _thread
+import time
+
+# return the sun_path of the most recently started program
+def get_sun_path(prog, port):
+
+    latest_path = ""
+    res = subprocess.run(["pgrep", "-l", prog], stdout = subprocess.PIPE)
+    # decode pgrep output from bytes and get the pids and program names
+    res_arr = res.stdout.decode("utf-8").split("\n")
+
+    for proc in res_arr:
+        if proc:
+            try:
+                tmp_arr = proc.split(" ")
+                if tmp_arr[1] == prog:
+                    latest_path = "%s/%s/%s" % (tmp_arr[0],prog,port)
+            except:
+                continue
+
+    return latest_path
+
+class Socket:
+
+    def __init__(self, prog, port):
+        self.m_prog = prog
+        self.m_port = port
+        self.m_connected = False
+        self.m_sock = socket.socket(socket.AF_UNIX, socket.SOCK_SEQPACKET)
+        self.m_sun_path = ""
+        self.m_partner = None
+
+    # add the connected socket
+    def add_partner(self, sock):
+        self.m_partner = sock
+
+    # connect to the c server socket
+    def connect(self):
+        if not self.m_connected:
+            sun_path = get_sun_path(self.m_prog, self.m_port)
+            try:
+                self.m_sock.connect("\0"+sun_path)
+                self.m_connected = True
+                self.m_sun_path = sun_path
+                self.setup_sock()
+                return True
+            except:
+                return False
+        return True
+
+    def setup_sock(self):
+        try:
+            _thread.start_new_thread(self.read, ())
+        except:
+            print("Error: unable to start read thread for %s." % (self.m_prog))
+
+    # read data from the c server socket and send to partner
+    def read(self):
+        while (self.m_connected):
+            data = self.m_sock.recv(25)
```

Comment:
Why 25?

The socket framework API uses packet boundaries to distinguish messages. If you send a message of size 100, what happens?